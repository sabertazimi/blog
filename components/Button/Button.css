/* Add dark mode support for antd Button component */
button.ant-btn.ant-btn-text {
  @apply dark:text-light dark:hover:text-gray-400;
}

button.ant-btn.ant-btn-default,
button.ant-btn.ant-btn-dashed {
  @apply bg-transparent hover:bg-transparent focus:bg-transparent dark:text-light dark:hover:text-secondary dark:focus:text-secondary;
}
