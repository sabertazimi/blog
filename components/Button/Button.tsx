import { Button as AntButton } from 'antd';
import type { ButtonProps } from 'antd';

interface Props extends ButtonProps {}

const Button = (props: Props): JSX.Element => <AntButton {...props} />;

export default Button;
