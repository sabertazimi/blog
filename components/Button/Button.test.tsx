import { render } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  test('should render correctly (snapshot)', () => {
    const { container } = render(<Button>Button</Button>);

    expect(container).toMatchSnapshot();
  });
});
