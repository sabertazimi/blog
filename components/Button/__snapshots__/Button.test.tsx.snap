// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly (snapshot) 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default"
    type="button"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;
